apps/teamsjs-diagnostic-app/src/components/sample/Welcome.tsx

```diff
@@ -0,0 +1,89 @@
+import { useContext, useState } from "react";
+import {
+  Image,
+  TabList,
+  Tab,
+  SelectTabEvent,
+  SelectTabData,
+  TabValue,
+} from "@fluentui/react-components";
+import "./Welcome.css";
+import { Scenario1 } from "./Scenario1";
+import { useData } from "@microsoft/teamsfx-react";
+import { TeamsJs } from "./TeamsJs";
+import { WebHub } from "./WebHub";
+import { TeamsFxContext } from "../Context";
+import { app } from "@microsoft/teams-js";
+
+export function Welcome(props: { showFunction?: boolean; environment?: string }) {
+  const { showFunction, environment } = {
+    showFunction: true,
+    environment: window.location.hostname === "localhost" ? "local" : "azure",
+    ...props,
+  };
+  const friendlyEnvironmentName =
+    {
+      local: "local environment",
+      azure: "Azure environment",
+    }[environment] || "local environment";
+
+  const [selectedValue, setSelectedValue] = useState<TabValue>("local");
+
+  const onTabSelect = (event: SelectTabEvent, data: SelectTabData) => {
+    setSelectedValue(data.value);
+  };
+  const { teamsUserCredential } = useContext(TeamsFxContext);
+  const { loading, data, error } = useData(async () => {
+    if (teamsUserCredential) {
+      const userInfo = await teamsUserCredential.getUserInfo();
+      return userInfo;
+    }
+  });
+  const userName = loading || error ? "" : data!.displayName;
```

Comment:
What if data is null? Do you think having a `!` non-null operator would break the app in runtime?

Reply:
This was a file from the tab app! But I will look into that

Reply:
When i remove it, it gives me the error that it's "possibly undefined"

Reply:
Fixed it!